shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -618,7 +618,19 @@ - (UITextRange*)rangeEnclosingPosition:(UITextPosition*)position
   return result;
 }
 
-- (UITextRange*)lineEnclosingPosition:(UITextPosition*)position {
+- (UITextRange*)lineEnclosingPosition:(UITextPosition*)position
+                          inDirection:(UITextDirection)direction {
+  // TODO(hellohuanlin): remove iOS 17 check. The same logic should apply to older iOS version.
+  if (@available(iOS 17.0, *)) {
+    // If querying the line from end of document in forward direction, report not found.
```

Comment:
This should link to and quote the docs I mentioned in my comment so that it's clear why.